e2e/testcases/ssh_known_hosts_test.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"kpt.dev/configsync/e2e/nomostest"
+	"kpt.dev/configsync/e2e/nomostest/ntopts"
+	nomostesting "kpt.dev/configsync/e2e/nomostest/testing"
+	"kpt.dev/configsync/e2e/nomostest/testpredicates"
+	v1 "kpt.dev/configsync/pkg/api/configmanagement/v1"
+	"kpt.dev/configsync/pkg/api/configsync"
+	"kpt.dev/configsync/pkg/core"
+	"kpt.dev/configsync/pkg/kinds"
+	"kpt.dev/configsync/pkg/reconcilermanager"
+	"kpt.dev/configsync/pkg/reconcilermanager/controllers"
+)
+
+func TestSSHKnownHost(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.SyncSource, ntopts.SkipNonLocalGitProvider, ntopts.NamespaceRepo(backendNamespace, configsync.RepoSyncName))
+	knownHostValue := "dir"
+	var err error
+
+	// create secret with known_hosts key
+	rootSecret := &corev1.Secret{}
+	err = nt.Validate(nomostest.RootAuthSecretName, v1.NSConfigManagementSystem, rootSecret)
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	err = nt.Watcher.WatchObject(kinds.Deployment(),
+		core.RootReconcilerPrefix, v1.NSConfigManagementSystem,
+		[]testpredicates.Predicate{
+			testpredicates.DeploymentHasEnvVar(reconcilermanager.GitSync, controllers.GitSyncKnownHosts, "false"),
+		},
+	)
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.MustMergePatch(rootSecret, secretDataPatch(controllers.KnownHostsKey, knownHostValue))
+
+	// validate known_hosts key in secret
+	require.NoError(nt.T,
```

Comment:
This seems a bit overkill, I think we can safely assume that if the patch was successful that the Secret was updated. There's no reconciliation happening here that we should need to verify, since the test it patching the Secret directly.

Reply:
sounds good, this part of validation has been removed